flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBStateBackend.java

```diff
@@ -236,8 +239,25 @@ public RocksDBStateBackend(StateBackend checkpointStreamBackend) {
 	 * @param enableIncrementalCheckpointing True if incremental checkpointing is enabled.
 	 */
 	public RocksDBStateBackend(StateBackend checkpointStreamBackend, TernaryBoolean enableIncrementalCheckpointing) {
+		this(checkpointStreamBackend, enableIncrementalCheckpointing, CheckpointingOptions.CHECKPOINT_RESTORE_THREAD_NUM.defaultValue());
+	}
+
+	/**
+	 * Creates a new {@code RocksDBStateBackend} that uses the given state backend to store its
+	 * checkpoint data streams. Typically, one would supply a filesystem or database state backend
+	 * here where the snapshots from RocksDB would be stored.
+	 *
+	 * <p>The snapshots of the RocksDB state will be stored using the given backend's
+	 * {@link StateBackend#createCheckpointStorage(JobID)}.
+	 *
+	 * @param checkpointStreamBackend The backend write the checkpoint streams to.
+	 * @param enableIncrementalCheckpointing True if incremental checkpointing is enabled.
+	 * @param restoredThreadNum thread num used to download files from DFS when restore.
+	 */
+	public RocksDBStateBackend(StateBackend checkpointStreamBackend, TernaryBoolean enableIncrementalCheckpointing, int restoredThreadNum) {
 		this.checkpointStreamBackend = checkNotNull(checkpointStreamBackend);
 		this.enableIncrementalCheckpointing = enableIncrementalCheckpointing;
+		this.restoredThreadNum = restoredThreadNum;
```

Comment:
Actually, instead of changing and exploding the constructor signatures, I would suggest to set it here to -1 (undefined). Then we could remove `final` from `restoredThreadNum` and add setter method for it, like for `predefinedOptions`. Users could use the setter to set this option per job.